### PR TITLE
DOC: fix more typos found by codespell

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
#### What does this implement/fix?

Fix more typos found by [codespell](https://github.com/codespell-project/codespell).

#### Additional information

I tried hard not to modify vendored files such as Lapack sources. Neverthless, please double-check as it is sometimes hard to identify if a file is vendored and might be updated upstream.

### CODE REVIEW

**Review by lucascolley — COMMENTED** [review]
1 comment, rest looks good

File: scipy/integrate/_quadpack_py.py [hunk 1]
```diff
@@ -519,7 +519,7 @@ def refunc(x, *args):
                 msg = ("If 'epsabs'<=0, 'epsrel' must be greater than both"
                        " 5e-29 and 50*(machine epsilon).")
             elif weight in ['sin', 'cos'] and (abs(a) + abs(b) == np.inf):
-                msg = ("Sine or cosine weighted intergals with infinite domain"
+                msg = ("Sine or cosine weighted intervals with infinite domain"
```
[the remark this answers is not in the document]
Comment (DimitriPapadopoulos) [reply scipy/integrate/_quadpack_py.py]: Since this is about computing integrals, I guess it's `integrals` :smile:

**Review by lucascolley — APPROVED** [review]
modulo whether `scipy/stats/biasedurn/wnchyppr.cpp` and `scipy/optimize/_direct/direct_wrap.c` should be modified or not.

**Comment by DimitriPapadopoulos** [comment]
* File `scipy/stats/biasedurn/wnchyppr.cpp` from [BiasedUrn](https://cran.r-project.org/package=BiasedUrn) should certainly not be changed, sorry about that.
* I thought that `direct_wrap.c` is a local addition to DIRECT, but it is not.

Will revert both.

**Review by j-bowhay — APPROVED** [review]
Thanks @DimitriPapadopoulos

File: scipy/interpolate/interpnd.pyx [hunk 2]
```diff
@@ -875,7 +875,7 @@ class CloughTocher2DInterpolator(NDInterpolatorBase):
     continuously differentiable.
 
     The gradients of the interpolant are chosen so that the curvature
-    of the interpolating surface is approximatively minimized. The
+    of the interpolating surface is approximately minimized. The
```
Comment (h-vetinari) [inline scipy/interpolate/interpnd.pyx:878]: This shouldn't have been changed, and we've already had that [discussion](https://github.com/scipy/scipy/pull/19041#discussion_r1335258728). The statement is not the same; one describes a process, the other a result. Please revert this change, and stop re-introducing it. Math sometimes uses language that's unusual in other contexts, and it's codespell that has to adapt to this, not math.

Comment (DimitriPapadopoulos) [reply scipy/interpolate/interpnd.pyx:878]: Sorry about that.

Comment (DimitriPapadopoulos) [reply scipy/interpolate/interpnd.pyx:878]: Would you agree to adding a codespell config in pyproject.toml and a file with codespell false positives?

Comment (h-vetinari) [reply scipy/interpolate/interpnd.pyx:878]: Thanks for fixing this quickly.

> Would you agree to adding a codespell config in pyproject.toml and a file with codespell false positives?

That depends on the other maintainers; I won't object, but I'm not enthusiastic about adding this to our pyproject.toml, which so far is quite strictly limited to project metadata and what's necessary for building wheels.

On top of that, my argument remains that accepting correct spellings is the responsibility codespell, and it shouldn't fall to us to carry "false positives" for stuff that codespell flags incorrectly. While I haven't invested time in finding out how codespell works under the hood, my understanding is that this would be fixed by telling codespell not to warn on words in the "rare" dictionary.

That sounds like the best solution to me _by far_, and so I'd ask you to either try that first, or provide an argument why it's not possible.

Comment (DimitriPapadopoulos) [reply scipy/interpolate/interpnd.pyx:878]: Nowadays it is certainly best to use spellchecking based on deep learning and a large corpus of N-grams. By the way, this would raise the question of how to split "words" in a more acute way than in most natural languages. Any way, codespell checks individual words, so we need to find a balance between too many false positives and too many false negatives. I feel it is hard to get absolutely no false positives, some manual adjustments might be necessary.
If you would rather keep `pyproject.toml` clean, the solution would be a `.codespellrc` file. As for the contents of the config, they could leave out the dictionary of typos that are rare words, or manually create a list of false positives to ignore.

Comment (lucascolley) [reply scipy/interpolate/interpnd.pyx:878]: You can write a proposal to [the mailing list](https://mail.python.org/archives/list/scipy-dev@python.org/) if you would like to. Do you envision this being a check in CI, or more (somewhat) regular PRs like this one? Either way, I imagine that at least some maintainers will object to it being a good use of time, but feel free to give your best argument if you are confident that it is worth it.
